boards: frdm_mcxa166, frdm_mcxa346, frdm_mcxa366: add lpdac support

peterwangsz opened the pull request:
1. enable nxp,lpdac
2. verified samples/drivers/dac
3. dac_out pin conflict with lpuart2 tx use lpuart3 in the samples/drivers/dac

peterwangsz commented:
hi @hakehuang , the DAC_OUT pin is conflict with LPUART2 TX, so I have to switch to LPUART3 in samples/drivers/dac.
The J5 Pin 3,4 are LPUART3 Rx,Tx pins. Please pay attention while setting up the CI test, thanks.
PS: there is only one DAC_OUT pin.

hakehuang reviewed (commented):
@peterwangsz , @NeilChen93  please add `dac` to board `support` list, besides how do you plan to handle the tests/drivers/dac?

maybe you can create a new test instance, with below fixture settings, then CI can use this fixture to deploy test cases
```
sample.drivers.dac:
  filter: CONFIG_DAC_SAMPLE_RUN and not dt_alias_exists("second_uart")
...
samples.dac.sec_uart:
  filter:  CONFIG_DAC_SAMPLE_RUN and dt_alias_exists("second_uart")
```

and add 

```
alias {
    second_uart = &lpuart3;
}
```

to those boards, which will help to filter out those platform, and I will add pytest script to check the second port afterwards

hakehuang reviewed (dismissed):
board testing pass for lpadc

hakehuang commented on tests/drivers/dac/dac_api/testcase.yaml:
```diff
@@ -6,4 +6,9 @@ common:
+    min_ram: 16
+  drivers.dac.api.sec_uart:
+    depends_on: dac
+    filter: dt_alias_exists("second_uart")
```
@martinjaeger, in this board primary UART is pinmux with adc, so we have to use a `second_uart` to avoid pin  conflicts. and other platform which not defined second_uart will not test the `drivers.dac.api.sec_uart`, as the filter will works.

peterwangsz commented on samples/drivers/dac/boards/frdm_mcxa266.overlay:
```diff
@@ -0,0 +1,33 @@
+/ {
+	alias {
+		second_uart = &lpuart3;
+	};
```
I agree, it is better. @NeilChen93 , for DAC sample and test case, please use lpuart3 for console and shell with board overlay, thanks.

martinjaeger reviewed (changes requested):
Looks good in general. Minor comment left.

martinjaeger commented on samples/drivers/dac/boards/frdm_mcxa266.overlay:
```diff
@@ -0,0 +1,25 @@
+
+&lpuart3 {
+	status = "okay";
+};
```
Ping @peterwangsz. Any comments?

martinjaeger commented:
@peterwangsz Can you please have another look?

peterwangsz commented:
hi @martinjaeger , I will update this PR this week.

peterwangsz committed:
boards: nxp: frdm_mcxaxx6: add lpdac support

Add DAC0 (LPDAC) support for frdm_mcxa266, frdm_mcxa346, and
frdm_mcxa366 boards.

DTS: add dac0 node to nxp_mcxaxx6_common.dtsi at address
0x400b4000, compatible "nxp,lpdac", IRQ 67.

Board: add pinmux_dac0 (DAC0_OUT_P2_2) to each board pinctrl
dtsi; enable dac0 with pinctrl in board_common.dtsi; set up DAC0
clock in board.c (SPC power-on, divider, attach, gate); add "dac"
to each board yaml supported-features list.

Sample/test: add board overlays for samples/drivers/dac and
tests/drivers/dac/dac_api. Because DAC0 output pin P2_2 conflicts
with LPUART2 TX, the overlays disable lpuart2 and redirect the
console to lpuart3. Add fixture: dac_fixture to testcase.yaml so
CI is aware of the special UART setup. Update the DAC sample README
with build instructions and the CONFIG_DAC_SAMPLE_RUN requirement.

DAC output is available on connector J1 pin 4 (P2_2).

Co-authored-by: Neil Chen <cheng.chen_1@nxp.com>
Signed-off-by: Peter Wang <chaoyi.wang@nxp.com>

peterwangsz commented:
> @peterwangsz Can you please have another look?
Hi @martinjaeger , I have updated the PR based on the comments, please help to review, thanks.

peterwangsz commented:
> as this need addition uart setting, for easy CI setting, can you add below to tests.yaml? Thanks. @peterwangsz
> 
> ```
> tests:
>   drivers.dac.api:
>     depends_on: dac
>     min_ram: 16
>     # some platform need special settings, add a fixture to notice CI settings
>     fixture: dac_fixture
> ```

Hi @hakehuang , updated in new commit, please help to review and test, thanks.

peterwangsz commented:
> > hi @hakehuang , the DAC_OUT pin is conflict with LPUART2 TX, so I have to switch to LPUART3 in samples/drivers/dac.
> > The J5 Pin 3,4 are LPUART3 Rx,Tx pins. Please pay attention while setting up the CI test, thanks.
> > PS: there is only one DAC_OUT pin.
> 
> as this need addition uart setting, for easy CI setting, can you add below to tests.yaml? Thanks. @peterwangsz
> 
> ```
> tests:
>   drivers.dac.api:
>     depends_on: dac
>     min_ram: 16
>     # some platform need special settings, add a fixture to notice CI settings
>     fixture: dac_fixture
> ```

hi @hakehuang, please help to test.
CI report failure https://github.com/zephyrproject-rtos/zephyr/actions/runs/28214278996/job/83582427248?pr=91338 that the "Additional properties are not allowed ('fixture' was unexpected)". 
Please review whether the tests/drivers/dac/dac_api/testcase.yaml update is correct or not, thanks.

hakehuang reviewed (approved):
board regression test pass, will add pytest enhancement for easy test

butok commented:
Hi @martinjaeger 
The PR has been updated,
please review and unblock if it addressed your comment.
